From fbd5a587cc144e298e30b3f9585e4d6c214bce89 Mon Sep 17 00:00:00 2001
From: Dhaval Shah <dhaval.shah@xilinx.com>
Date: Mon, 13 Jan 2020 00:20:43 -0800
Subject: [PATCH 0805/1851] soc: xilinx: vcu: Provided API to get core clock

commit 5130d97b2c46ff0472f6f30113ef5fde4aaa9d88 from
https://github.com/Xilinx/linux-xlnx.git

Core Clock frequency related information is provided
through the xvcu_get_clock_frequency API.

Signed-off-by: Dhaval Shah <dhaval.shah@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/soc/xilinx/xlnx_vcu.c | 13 +++++++++++++
 include/soc/xilinx/xlnx_vcu.h |  1 +
 2 files changed, 14 insertions(+)

diff --git a/drivers/soc/xilinx/xlnx_vcu.c b/drivers/soc/xilinx/xlnx_vcu.c
index 9216b4142f16..3173c822e09a 100644
--- a/drivers/soc/xilinx/xlnx_vcu.c
+++ b/drivers/soc/xilinx/xlnx_vcu.c
@@ -304,6 +304,19 @@ u32 xvcu_get_memory_depth(struct xvcu_device *xvcu)
 }
 EXPORT_SYMBOL_GPL(xvcu_get_memory_depth);
 
+/**
+ * xvcu_get_clock_frequency - provide the core clock frequency
+ * @xvcu:	Pointer to the xvcu_device structure
+ *
+ * Return:	Returns 32bit value
+ *
+ */
+u32 xvcu_get_clock_frequency(struct xvcu_device *xvcu)
+{
+	return xvcu->coreclk;
+}
+EXPORT_SYMBOL_GPL(xvcu_get_clock_frequency);
+
 /**
  * xvcu_set_vcu_pll_info - Set the VCU PLL info
  * @xvcu:	Pointer to the xvcu_device structure
diff --git a/include/soc/xilinx/xlnx_vcu.h b/include/soc/xilinx/xlnx_vcu.h
index ca3333361c0d..c81c757eae68 100644
--- a/include/soc/xilinx/xlnx_vcu.h
+++ b/include/soc/xilinx/xlnx_vcu.h
@@ -14,5 +14,6 @@ struct xvcu_device;
 
 u32 xvcu_get_color_depth(struct xvcu_device *xvcu);
 u32 xvcu_get_memory_depth(struct xvcu_device *xvcu);
+u32 xvcu_get_clock_frequency(struct xvcu_device *xvcu);
 
 #endif  /* _XILINX_VCU_H_ */
-- 
2.31.1

